docs/analyse.md

```diff
@@ -40,10 +40,273 @@ watch --interval 30 python -m neps.status ROOT_DIRECTORY
+
+#### The Logging Function
+
+The `tblogger.log` function is invoked within the model's training loop to facilitate logging of key metrics.
+
+- **Signature:**
+```python
+tblogger.log(
+    loss: float,
+    current_epoch: int,
+    write_summary_incumbent: bool = False,
+    write_config_scalar: bool = False,
+    write_config_hparam: bool = True,
+    extra_data: dict | None = None
+)
+```
+
+- **Parameters:**
+    - `loss` (float): The loss value to be logged.
+    - `current_epoch` (int): The current epoch or iteration number.
+    - `write_summary_incumbent` (bool, optional): Set to `True` for a live incumbent trajectory.
+    - `write_config_scalar` (bool, optional): Set to `True` for a live loss trajectory for each configuration.
+    - `write_config_hparam` (bool, optional): Set to `True` for live parallel coordinate, scatter plot matrix, and table view.
+    - `extra_data` (dict, optional): Additional data to be logged, provided as a dictionary.
+
+#### Extra Custom Logging
+
+NePS provides dedicated functions for customized logging using the `extra_data` argument. 
+Note: Name the dictionary keys as the names of the values you want to log and pass one of the following functions as the values for a successful logging process.
+
+##### 1- Extra Scalar Logging
+
+Logs new scalar data during training. Uses `current_epoch` from the log function as its `global_step`.
+
+- **Signature:**
+```python
+tblogger.scalar_logging(value: float)
+```
+- **Parameters:**
+    - `value` (float): Any scalar value to be logged at the current epoch of `tblogger.log` function.
+
+##### 2- Extra Image Logging
+
+Logs images during training. Images can be resized, randomly selected, and a specified number can be logged at specified intervals. Uses `current_epoch` from the log function as its `global_step`.
+
+- **Signature:**
+```python
+tblogger.image_logging(
+    image: torch.Tensor,
+    counter: int = 1,
+    resize_images: list[None | int] | None = None,
+    random_images: bool = True,
+    num_images: int = 20,
+    seed: int | np.random.RandomState | None = None,
+)
+```
+
+- **Parameters:**
+    - `image` (torch.Tensor): Image tensor to be logged.
+    - `counter` (int): Log images every counter epochs (i.e., when current_epoch % counter equals 0).
+    - `resize_images` (list of int, optional): List of integers for image sizes after resizing (default: [32, 32]).
+    - `random_images` (bool, optional): Images are randomly selected if True (default: True).
+    - `num_images` (int, optional): Number of images to log (default: 20).
+    - `seed` (int or np.random.RandomState or None, optional): Seed value or RandomState instance to control randomness and reproducibility (default: None).
+
+#### MNIST Example
```

Comment:
This section here is likely better to live in some `logging_tensorboard_example.py` and then importing it into the markdown with [this](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#embedding-external-files).

````
```title="Logging with tensorboard in NePS"
--8<-- "path/to/logging_tensorboard_example.py"
```
````

Reply:
Currently I did it so because it is more user friendly with less details than this one [here](https://github.com/automl/neps/blob/master/neps_examples/convenience/neps_tblogger_tutorial.py). However, I am open to changing the original example to the new one, knowing that the main purpose of the example is to showcase TensorBoard integration in NePS.